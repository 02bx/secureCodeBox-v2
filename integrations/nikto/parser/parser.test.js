const fs = require('fs');
const util = require('util');

// eslint-disable-next-line security/detect-non-literal-fs-filename
const readFile = util.promisify(fs.readFile);

const { parse } = require('./parser');

test('parses www.securecodebox.io result file into findings', async () => {
  const fileContent = JSON.parse(
    await readFile(__dirname + '/__testFiles__/www.securecodebox.io.json', {
      encoding: 'utf8',
    })
  );

  expect(await parse(fileContent)).toMatchSnapshot();
});
