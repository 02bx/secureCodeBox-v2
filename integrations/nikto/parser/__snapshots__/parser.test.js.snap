// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parses www.securecodebox.io result file into findings 1`] = `
Array [
  Object {
    "attributes": Object {
      "banner": "GitHub.com",
      "hostname": "www.securecodebox.io",
      "ip_address": "185.199.108.153",
      "method": "GET",
      "niktoId": 999986,
      "port": 443,
    },
    "category": "Nikto Finding",
    "description": null,
    "location": "https://www.securecodebox.io/",
    "name": "Retrieved via header: 1.1 varnish",
    "osi_layer": "NETWORK",
    "severity": "INFORMATIONAL",
  },
  Object {
    "attributes": Object {
      "banner": "GitHub.com",
      "hostname": "www.securecodebox.io",
      "ip_address": "185.199.108.153",
      "method": "GET",
      "niktoId": 999986,
      "port": 443,
    },
    "category": "Nikto Finding",
    "description": null,
    "location": "https://www.securecodebox.io/",
    "name": "Retrieved x-served-by header: cache-fra19151-FRA",
    "osi_layer": "NETWORK",
    "severity": "INFORMATIONAL",
  },
  Object {
    "attributes": Object {
      "banner": "GitHub.com",
      "hostname": "www.securecodebox.io",
      "ip_address": "185.199.108.153",
      "method": "GET",
      "niktoId": 999986,
      "port": 443,
    },
    "category": "Nikto Finding",
    "description": null,
    "location": "https://www.securecodebox.io/",
    "name": "Retrieved access-control-allow-origin header: *",
    "osi_layer": "NETWORK",
    "severity": "INFORMATIONAL",
  },
  Object {
    "attributes": Object {
      "banner": "GitHub.com",
      "hostname": "www.securecodebox.io",
      "ip_address": "185.199.108.153",
      "method": "GET",
      "niktoId": 999957,
      "port": 443,
    },
    "category": "X-Frame-Options Header",
    "description": null,
    "location": "https://www.securecodebox.io/",
    "name": "The anti-clickjacking X-Frame-Options header is not present.",
    "osi_layer": "NETWORK",
    "severity": "LOW",
  },
  Object {
    "attributes": Object {
      "banner": "GitHub.com",
      "hostname": "www.securecodebox.io",
      "ip_address": "185.199.108.153",
      "method": "GET",
      "niktoId": 999102,
      "port": 443,
    },
    "category": "X-XSS-Protection",
    "description": null,
    "location": "https://www.securecodebox.io/",
    "name": "The X-XSS-Protection header is not defined. This header can hint to the user agent to protect against some forms of XSS",
    "osi_layer": "NETWORK",
    "severity": "LOW",
  },
  Object {
    "attributes": Object {
      "banner": "GitHub.com",
      "hostname": "www.securecodebox.io",
      "ip_address": "185.199.108.153",
      "method": "GET",
      "niktoId": 999100,
      "port": 443,
    },
    "category": "Uncommon Header",
    "description": null,
    "location": "https://www.securecodebox.io/",
    "name": "Uncommon header 'x-timer' found, with contents: S1585519074.290715,VS0,VE1",
    "osi_layer": "NETWORK",
    "severity": "INFORMATIONAL",
  },
  Object {
    "attributes": Object {
      "banner": "GitHub.com",
      "hostname": "www.securecodebox.io",
      "ip_address": "185.199.108.153",
      "method": "GET",
      "niktoId": 999100,
      "port": 443,
    },
    "category": "Uncommon Header",
    "description": null,
    "location": "https://www.securecodebox.io/",
    "name": "Uncommon header 'x-cache' found, with contents: HIT",
    "osi_layer": "NETWORK",
    "severity": "INFORMATIONAL",
  },
  Object {
    "attributes": Object {
      "banner": "GitHub.com",
      "hostname": "www.securecodebox.io",
      "ip_address": "185.199.108.153",
      "method": "GET",
      "niktoId": 999100,
      "port": 443,
    },
    "category": "Uncommon Header",
    "description": null,
    "location": "https://www.securecodebox.io/",
    "name": "Uncommon header 'x-proxy-cache' found, with contents: MISS",
    "osi_layer": "NETWORK",
    "severity": "INFORMATIONAL",
  },
  Object {
    "attributes": Object {
      "banner": "GitHub.com",
      "hostname": "www.securecodebox.io",
      "ip_address": "185.199.108.153",
      "method": "GET",
      "niktoId": 999100,
      "port": 443,
    },
    "category": "Uncommon Header",
    "description": null,
    "location": "https://www.securecodebox.io/",
    "name": "Uncommon header 'x-fastly-request-id' found, with contents: 12821df5c3f5eb828b1a4ce7d4e3637faa71291a",
    "osi_layer": "NETWORK",
    "severity": "INFORMATIONAL",
  },
  Object {
    "attributes": Object {
      "banner": "GitHub.com",
      "hostname": "www.securecodebox.io",
      "ip_address": "185.199.108.153",
      "method": "GET",
      "niktoId": 999100,
      "port": 443,
    },
    "category": "Uncommon Header",
    "description": null,
    "location": "https://www.securecodebox.io/",
    "name": "Uncommon header 'x-github-request-id' found, with contents: DB72:2841:1B0932:23C885:5E8119E0",
    "osi_layer": "NETWORK",
    "severity": "INFORMATIONAL",
  },
  Object {
    "attributes": Object {
      "banner": "GitHub.com",
      "hostname": "www.securecodebox.io",
      "ip_address": "185.199.108.153",
      "method": "GET",
      "niktoId": 999100,
      "port": 443,
    },
    "category": "Uncommon Header",
    "description": null,
    "location": "https://www.securecodebox.io/",
    "name": "Uncommon header 'x-served-by' found, with contents: cache-fra19151-FRA",
    "osi_layer": "NETWORK",
    "severity": "INFORMATIONAL",
  },
  Object {
    "attributes": Object {
      "banner": "GitHub.com",
      "hostname": "www.securecodebox.io",
      "ip_address": "185.199.108.153",
      "method": "GET",
      "niktoId": 999955,
      "port": 443,
    },
    "category": "Nikto Finding",
    "description": null,
    "location": "https://www.securecodebox.io/",
    "name": "The site uses SSL and Expect-CT header is not present.",
    "osi_layer": "NETWORK",
    "severity": "INFORMATIONAL",
  },
  Object {
    "attributes": Object {
      "banner": "GitHub.com",
      "hostname": "www.securecodebox.io",
      "ip_address": "185.199.108.153",
      "method": "GET",
      "niktoId": 999103,
      "port": 443,
    },
    "category": "X-Content-Type-Options Header",
    "description": null,
    "location": "https://www.securecodebox.io/",
    "name": "The X-Content-Type-Options header is not set. This could allow the user agent to render the content of the site in a different fashion to the MIME type",
    "osi_layer": "NETWORK",
    "severity": "INFORMATIONAL",
  },
  Object {
    "attributes": Object {
      "banner": "GitHub.com",
      "hostname": "www.securecodebox.io",
      "ip_address": "185.199.108.153",
      "method": "PUT",
      "niktoId": 999962,
      "port": 443,
    },
    "category": "Nikto Finding",
    "description": null,
    "location": "https://www.securecodebox.io/",
    "name": "Server banner changed from 'GitHub.com' to 'Varnish'",
    "osi_layer": "NETWORK",
    "severity": "INFORMATIONAL",
  },
  Object {
    "attributes": Object {
      "banner": "GitHub.com",
      "hostname": "www.securecodebox.io",
      "ip_address": "185.199.108.153",
      "method": "GET",
      "niktoId": 999966,
      "port": 443,
    },
    "category": "Nikto Finding",
    "description": null,
    "location": "https://www.securecodebox.io/",
    "name": "The Content-Encoding header is set to \\"deflate\\" this may mean that the server is vulnerable to the BREACH attack.",
    "osi_layer": "NETWORK",
    "severity": "INFORMATIONAL",
  },
]
`;
